app/controllers/pets_controller.rb

```diff
@@ -13,16 +13,9 @@ def new
   end
 
   def create
-    @pet = Pet.create({
-      name: params[:name],
-      approximate_age: params[:approximate_age],
-      sex: params[:sex],
-      image_path: params[:image_path],
-      description: params[:description],
-      adoption_status: "Adoptable",
-      shelter_id: params[:shelter_id]
-      })
-    redirect_to "/shelters/#{@pet.shelter_id}/pets"
+    shelter = Shelter.find(params[:shelter_id])
+    shelter.pets.create!(pet_params)
+    redirect_to "/shelters/#{shelter.id}/pets"
```

Comment:
refactored to use a private action to gather the params and create! so I don't have to use an instance variable.